cwl/src/main/scala/cwl/InitialWorkDirRequirement.scala

```diff
@@ -0,0 +1,70 @@
+package cwl
+
+import cwl.InitialWorkDirRequirement.{ExpressionOrString, IwdrListingArrayEntry, _}
+import eu.timepit.refined.W
+import shapeless.{:+:, CNil, _}
+
+final case class InitialWorkDirRequirement(
+                                            `class`: W.`"InitialWorkDirRequirement"`.T,
+                                            listing: Array[IwdrListingArrayEntry] :+: ExpressionOrString :+: CNil
+                                          ) {
+  val listings: Array[IwdrListingArrayEntry] = listing match {
+    case IwdrListingArray(array) => array
+    case ExpressionOrString(eos) => Array(Coproduct[IwdrListingArrayEntry](eos))
+  }
+
+  override def toString: WorkflowStepInputId =
+    s"""InitialWorkDirRequirement(
+      |  ${listings.mkString(System.lineSeparator + "  ")}
+      |)""".stripMargin
+}
+
+/**
+  *  Short for "Directory Entry"
+  *  @see <a href="http://www.commonwl.org/v1.0/CommandLineTool.html#Dirent">Dirent Specification</a>
+  *
+  *  Split into two cases because entryName is only optional if entry is an Expression
+  */
+trait Dirent {
+  def writable: Option[Boolean]
```

Comment:
consider making this `protected`?  Seems like one would always want `writableWithDefault`.  

Although off the top of my head I don't know how the override w/ case class would work, it might anger the compiler if it made the field public.